pipelines/services/syncstatus.go

```diff
@@ -0,0 +1,122 @@
+package services
+
+import (
+	"encoding/json"
+	"github.com/jfrog/jfrog-client-go/auth"
+	"github.com/jfrog/jfrog-client-go/http/jfroghttpclient"
+	"github.com/jfrog/jfrog-client-go/utils/errorutils"
+	"github.com/jfrog/jfrog-client-go/utils/io/httputils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"net/http"
+	"net/url"
+	"strconv"
+)
+
+type SyncStatusService struct {
+	client *jfroghttpclient.JfrogHttpClient
+	auth.ServiceDetails
+}
+
+func (ss *SyncStatusService) getHttpDetails() httputils.HttpClientDetails {
+	httpDetails := ss.ServiceDetails.CreateHttpClientDetails()
+	return httpDetails
+}
+
+func NewSyncStatusService(client *jfroghttpclient.JfrogHttpClient) *SyncStatusService {
+	return &SyncStatusService{client: client}
+}
+
+// GetSyncPipelineResourceStatus fetches pipeline sync status
+func (ss *SyncStatusService) GetSyncPipelineResourceStatus(repoName, branch string) ([]PipelineSyncStatus, []byte, error) {
```

Comment:
Let's not return the body
```suggestion
func (ss *SyncStatusService) GetSyncPipelineResourceStatus(repoName, branch string) ([]PipelineSyncStatus, error) {
```